### ISSUE TITLE
Unsupported GNU_PROPERTY_TYPE and drop in frame rates

### ISSUE THREAD

**Issue by Feichtmeier** [canonical/flutter-snap#44]
Hello there :wave: 

I receive this warning when starting a flutter app with flutter installed from snap since the update to 0+git.e75662a 2021-04-15
```
/snap/flutter/current/usr/bin/ld: warning: /usr/lib/gcc/x86_64-linux-gnu/10/../../../x86_64-linux-gnu/crt1.o: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0008002
```
It is probably unrelated but I also noticed that the whole performance has dropped since 0+git.e75662a 2021-04-15 and some frames are missed, it looks like the window is lagging

![](https://user-images.githubusercontent.com/15329494/115574886-53e7fe00-a2c2-11eb-9c04-319d9bb1647f.gif)

It is not much information I can provide, sorry.

**Comment by Feichtmeier** [comment]
@MarcusTomlinson the performance is a little bit better with flutter version 2.3.0-1.0.pre but still progress indicators are lagging 

This warning also won't go away after upgrading the flutter snap
```
/snap/flutter/current/usr/bin/ld: warning: /usr/lib/gcc/x86_64-linux-gnu/10/../../../x86_64-linux-gnu/crt1.o: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0008002
/snap/flutter/current/usr/bin/ld: warning: //lib/x86_64-linux-gnu/libdl.so.2: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0008002
```

Any idea why this is happening and if the performance is related to those warnings?
On MacOS the performance has no issues and those warnings (ofc...) don't appear (no snap :) )

**Comment by MarcusTomlinson** [comment]
The performance issues are likely due to us forcing software rendering with `flutter run` via the snap:

https://github.com/canonical/flutter-snap/blob/master/env.sh#L19

This won't effect the actual build though. i.e. running the binary from the `build/linux/x64/blah/bundle/`

**Comment by Feichtmeier** [comment]
Hm, I ran it from the build ... dir and it is still lagging sometimes, then it goes back to 60 fps
![Peek 2021-05-12 10-13](https://user-images.githubusercontent.com/15329494/117941904-cf116280-b30a-11eb-8311-a36a524de94f.gif)

**Comment by MarcusTomlinson** [comment]
Are you running a debug or a release build?

**Comment by Feichtmeier** [comment]
Release I think
![grafik](https://user-images.githubusercontent.com/15329494/117944262-5e1f7a00-b30d-11eb-9f2b-771f92e3ee67.png)

**Comment by Feichtmeier** [comment]
https://user-images.githubusercontent.com/15329494/117951154-2c5de180-b314-11eb-9bfc-96e50c4e0b6b.mp4

Left is the "linux binary?" run from the bundle and right is chrome run from main.dart in vscode
Ofc the framerate is not optimal when recording but you can clearly see the lags in the linux binary :)

**Comment by Feichtmeier** [comment]
@MarcusTomlinson I witched to the edge channel of the flutter snap and now the lags are gone :open_mouth: 

![Peek 2021-05-19 15-22](https://user-images.githubusercontent.com/15329494/118819953-1cb33f80-b8b6-11eb-8cb8-24af6e041568.gif)

**Comment by Feichtmeier** [comment]
It is still below macos desktop performance, but still good. The bigger I make the window, the more the linear progress indicator lags, but still performance is good.

https://user-images.githubusercontent.com/15329494/118821295-62bcd300-b8b7-11eb-94ef-842d7467b6f3.mp4

**Comment by Feichtmeier** [comment]
Alright @MarcusTomlinson I am finally kind of sure what the reason for all those lags are.
It seems that flutter apps are really dependent on the cpu freq covernor.
I used a snap called [auto cpu freq](https://snapcraft.io/auto-cpufreq) which optimizes the cpu freq. With this running there was always 1 interval with lags and then 1 interval without. After uninstalling it, and relying totally on the gnome controler center setting for the cpu covernor, I can clearly see that I am at high fps with full system power, lower fps at mid power and very low fps and low power setting - and this was recorded with the started binary from example/build/linux/x64/debug/bundle : 

![cpu](https://user-images.githubusercontent.com/15329494/122519177-73757b80-d012-11eb-83a6-27889b66ccb3.gif)

So this is prbly unrelated to those unsupported gnue property type warnings, right? :)

**Comment by MarcusTomlinson** [comment]
@Feichtmeier do you think we could close this then?

**Comment by Feichtmeier** [comment]
@MarcusTomlinson initially I only created the issue for the warning. I think it still exists, but I guess if it is no real issue we can close it
The performance is not caused by snap but by the CPU covernor

**Comment by MarcusTomlinson** [comment]
> @MarcusTomlinson initially I only created the issue for the warning. I think it still exists, but I guess if it is no real issue we can close it
> The performance is not caused by snap but by the CPU covernor

Yeah, those warnings have been around for quite some time and so far appear to be harmless. I shall close, thanks.
